gpiocontrol/source/objc/GPIOControlDemo/DevicesTableViewController.m

```diff
@@ -172,24 +102,26 @@ - (BOOL)tableView:(UITableView *)tableView canEditRowAtIndexPath:(NSIndexPath *)
 
 - (void)tableView:(UITableView *)tableView commitEditingStyle:(UITableViewCellEditingStyle)editingStyle forRowAtIndexPath:(NSIndexPath *)indexPath {
     if (editingStyle == UITableViewCellEditingStyleDelete) {
-        id<KaaClient> kaaClient = [KaaProvider getClient];
-        NSLog(@"%@ Going to detach.", TAG);
         Device *currentDevice = self.devices[indexPath.row];
         self.deletingIndexPath = indexPath;
-        [kaaClient detachEndpointWithKeyHash:[[EndpointKeyHash alloc] initWithKeyHash:currentDevice.kaaEndpointId] delegate:self];
+        [[KaaClientManager sharedManager] detachEndpoint:currentDevice.kaaEndpointId
+                                                delegate:self];
     }
 }
 
-#pragma mark - Delegates methods
+#pragma mark - Kaa Delegates methods
```

Comment:
It's better to call mark with corresponding protocol name(e.g. RemoteControlECFDelegate, OnDetachEndpointOperationDelegate, OnAttachEndpointOperationDelegate) for every methods block.